test/Raphael/raphael.test.ts

```diff
@@ -224,6 +224,41 @@ describe("Raphael DAO contract", () => {
                     expect(votesAgainst.eq(ethers.constants.Zero))
                 });
 
+                // tests vote()
+                it("getDidIVote returns correct values", async () => {
+                    // Should start with 0 proposals
+                    expect(await raphael.proposalCount()).to.equal(0);
+                    let tx = await (await raphael.connect(admin).createProposal("Proposal n details")).wait()
+                    const filteredEvents = tx?.events.filter((eventItem: any) => eventItem.event === "ProposalCreated")
+                        .map((eventItem: any) => eventItem.args)
+                    const { proposalId, vote_start } = filteredEvents[0];
+
+                    let blocks_skipped = vote_start.sub(BigNumber.from(tx.blockNumber))
+
+                    await skipBlocks(blocks_skipped)
+
+                    // updateProposalStatus to 1 = VOTING
+                    tx = await (await raphael.connect(admin).updateProposalStatus(proposalId)).wait()
+                    let status = (await raphael.getProposalData(proposalId))[5]
+                    expect(status).to.equal((BigNumber.from(PROPOSAL_STATUS.VOTING))) // now in voting period
+
+                    
+
+                    // check user's balance (voting power)
+                    let userStakedBalance = await staking.getStakedBalance(userAddress)
+                    // user votes true (for)
+                    await raphael.connect(user).vote(proposalId, true)
+                    // check current votes for and against proposal
+                    let proposalData = await raphael.getProposalData(proposalId)
+                    let votesFor = proposalData[1]
+                    let votesAgainst = proposalData[2]
+
+                    // expect votesFor to be user's token balance
+                    expect(votesFor).to.equal(userStakedBalance);
+                    // expect no votes against
+                    expect(votesAgainst.eq(ethers.constants.Zero))
+                });
+
```

Comment:
you may also want to test that it returns false when it should (when an address has not voted)

Reply:
This is addressed in the new commit.